packages/blue-sdk/src/maths/MathLib.ts

```diff
@@ -190,8 +190,9 @@ export class MathLib {
    * @param apr The rate to convert (in WAD)
    * @param compounding The compounding basis
    */
-  static rateToApy(rate: BigIntish, unit: Time.Unit) {
-    const factor = Time[unit].from.y(1n);
+  static rateToApy(rate: BigIntish, period: Time.PeriodLike) {
+    const { unit, duration } = Time.toPeriod(period);
+    const factor = Time[unit].from.y(1n) / BigInt(duration);
 
     return (
       (1 + Number(format.number.of(BigInt(rate), 18))) ** Number(factor) - 1
```

Comment:
```suggestion
    const factor = Time[unit].from.y(1) / duration;

    return (
      (1 + Number(format.number.of(BigInt(rate), 18))) ** factor - 1
```

Allows to have more precise calculations